### ISSUE TITLE
Extension Points: allow setting preset arguments at registration time

### ISSUE THREAD

**Issue by leonardder** [nvaccess/nvda#9852]
### Is your feature request related to a problem? Please describe.
In the current situation, we can't provide arguments to extension points at registration time that are implicitly send to the handler at notification time. Such a feature would come in handy for the vision framework (#9064).

### Describe the solution you'd like
My proposal is probably best described with a code example:

```
handleFocusChange = extensionPoints.Action()
handleCaretChange = extensionPoints.Action()

def handleChange(origin):
	print(origin)

handleFocusChange.register(handleChange, origin="focus")
handleCaretChange.register(handleChange, origin="caret")

>> handleFocusChange.notify()
focus
>> handleCaretChange.notify()
caret
```

In the current situation, this code would require a separate handler for every extension point.

In short, when having multiple extension points that require a similar action to be performed, it requires less code to reuse a single handler that is aware to handle information from all extension points.

### Describe alternatives you've considered
call the actions like this: `handleFocusChange.notify(origin="focus"`

**Comment by feerrenrut** [comment]
What about using a `partial` for this, or a `lambda`? I think I would prefer that rather than introducing more complexity into extension points.

**Comment by leonardder** [comment]
My idea was to use partials inside extension points, i.e. if you provide kwargs to a register call, automatically wrap the handler inside a partial that automatically sets the given kwargs. However, now I think about it, we need to save a reference to these partials anyway.

**Comment by leonardder** [comment]
After thinking more about it, I think partials are indeed the solution to this issue.
